File: payroll-api/src/main/java/com/company/payroll/mapper/SalaryMapper.java
package com.company.payroll.mapper;

import java.util.List;

import org.springframework.stereotype.Repository;

import com.company.payroll.model.Salary;

@Repository
public interface SalaryMapper {
    int deleteByPrimaryKey(Integer sId);

    int insert(Salary row);

    int insertSelective(Salary row);
    
    List<Salary> selectList();

    Salary selectByPrimaryKey(Integer sId);

    int updateByPrimaryKeySelective(Salary row);

    int updateByPrimaryKey(Salary row);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/SalaryMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.Salary
import org.springframework.stereotype.Repository

@Repository
interface SalaryMapper {
  fun deleteByPrimaryKey(sId: Int): Int
  fun insert(row: Salary): Int
  fun insertSelective(row: Salary): Int
  fun selectList(): List<Salary>?
  fun selectByPrimaryKey(sId: Int): Salary?
  fun updateByPrimaryKeySelective(row: Salary): Int
  fun updateByPrimaryKey(row: Salary): Int
}